### PR TITLE
feat: externalize RPC client scope configuration to file (closes #2119)

### LINKED ISSUE #2119: Make RPC Client Definitions Configurable at Build Time
## Description
We need to refactor our RPC client parsing logic to make client definitions configurable at build time or based on an external file rather than being hardcoded in the source code.

## Background
Currently, our client parsing logic is hardcoded in [src/eth/rpc/rpc_client_app.rs (lines 24-65)](https://github.com/cloudwalk/stratus/blob/main/src/eth/rpc/rpc_client_app.rs#L24-L65). This approach has several limitations:
- Adding or modifying client definitions requires code changes;
- It's difficult to maintain as the number of clients grows;
- As an open source solution, hardcoded values are not useful.

## Tasks
Design a configuration file format (e.g., JSON, YAML, TXT) to define RPC clients
Implement a parser to read client definitions from this file at build time
Refactor `src/eth/rpc/rpc_client_app.rs` to use the parsed configurations

### PR DESCRIPTION
**Pull request by bronzelle-cw**
## Summary
 
This PR implements a configurable RPC client scope system by externalizing hardcoded scope matching logic into a configuration file.

### 🔧 **Configuration File**
- Added `src/eth/rpc/client_scopes.txt` - External configuration file for client scope rules
- Supports 3 pattern types:
  - **Exact match**: `infra: blockscout` → `blockscout` becomes `infra::blockscout`
  - **Prefix match**: `issuing: issuing*` → `issuing-api` becomes `issuing::issuing-api`  
  - **Prefix trim**: `user: user-/` → `user-alice` becomes `user::alice`

### ⚙️ **Build-time Code Generation**
- Enhanced `build.rs` to parse the config file and generate Rust matching code
- Automatically rebuilds when `client_scopes.txt` changes
- Generated code is included in `rpc_client_app.rs`

### 🧹 **Code Cleanup**
- Replaced hardcoded scope matching logic with generated function
- Maintained exact same functionality while improving maintainability
- Added comprehensive test suite with edge case coverage

## Benefits

- **Maintainability**: Client scope rules can be updated without code changes
- **Clarity**: Configuration is now clearly documented with examples
- **Performance**: Generated code is as fast as hardcoded logic
- **Safety**: Build-time validation ensures configuration correctness

## Testing

Added extensive tests covering:
- All existing scope patterns
- Edge cases for exact matches
- Prefix matching behavior
- Fallback to 'other' scope

Closes #2119

___

### **PR Type**
Enhancement, Refactor

___

### **Description**
- Externalize RPC client scope configuration to file

- Implement build-time code generation for client scopes

- Add comprehensive test suite for scope parsing

- Refactor `RpcClientApp::parse` to use generated matcher

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["client_scopes.txt"] --> B["build.rs"]
  B --> C["Generated Matcher"]
  C --> D["rpc_client_app.rs"]
  D --> E["RpcClientApp::parse"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>build.rs</strong><dd><code>Implement build-time code generation for client scopes</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

build.rs

<ul><li>Add <code>generate_client_scopes_matcher()</code> function<br> <li> Implement parsing of <code>client_scopes.txt</code><br> <li> Generate Rust code for client scope matching</ul>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/2265/files#diff-d0d98998092552a1d3259338c2c71e118a5b8343dd4703c0c7f552ada7f9cb42">+70/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Refactor</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>rpc_client_app.rs</strong><dd><code>Refactor RpcClientApp to use generated scope matcher</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/rpc/rpc_client_app.rs

<ul><li>Include generated client scopes matcher<br> <li> Refactor <code>RpcClientApp::parse</code> to use generated matcher<br> <li> Add comprehensive test suite for scope parsing</ul>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/2265/files#diff-58a4136b4354d7489c12136de37282c85be59f542ed4d605ceead0c00ba5ddc3">+65/-33</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>client_scopes.txt</strong><dd><code>Add external configuration file for client scopes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/eth/rpc/client_scopes.txt

<ul><li>Add new configuration file for client scopes<br> <li> Define scope patterns for various client types<br> <li> Include comments explaining pattern types</ul>

</details>

  </td>
  <td><a href="https://github.com/cloudwalk/stratus/pull/2265/files#diff-8653ab3a5962d37692dac489a519a07c188ceea0fdb913c71f7ac660899cc713">+14/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: src/eth/rpc/client_scopes.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# prefix match:   `issuing: issuing*` → `issuing-api` becomes `issuing::issuing-api`  
+# prefix trim:    `user: user-/` → `user-alice` becomes `user::alice`
+
+stratus: stratus/
+acquiring: authorizer
+banking: balance* banking*
+issuing: issuing* infinitecard*
+lending: lending*
+infra: blockscout golani tx-replayer
+user: user-/ insomnia
```
Comment (gabriel-aranha-cw) [inline src/eth/rpc/client_scopes.txt:14]: I think you can move this file to static/client_scopes? to maintain these types of files under the same directory

**Review by gabriel-aranha-cw — CHANGES_REQUESTED** [review]
You can run `just lint` to automatically lint the code.

**Review by gabriel-aranha-cw — APPROVED** [review]
